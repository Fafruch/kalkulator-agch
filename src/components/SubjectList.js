/**
 * Created by Fafruch on 17/07/2017.
 */

import React from 'react'
import PropTypes from 'prop-types'
import Subject from './Subject'

const SubjectList = ({ subjects, onChange, onToggleClick, onDelete, subjectsType }) => (
  <ul>
    {subjects.map(subject =>
      subject.type === subjectsType ?
        <li key={subject.id}> {/* tylko dla iteracji, react tego potrzebuje */}
          <Subject
            id={subject.id}
            name={subject.name}
            elementaryScore={subject.elementaryScore}
            advancedScore={subject.advancedScore}
            active={subject.active}
            onChange={onChange}
            onToggleClick={onToggleClick}
            onDelete={onDelete}
            subjectsType={subjectsType}
          />
        </li>
        : null
    )}
  </ul>
)

SubjectList.propTypes = {
  subjects: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string,
    elementaryScore: PropTypes.number.isRequired,
    advancedScore: PropTypes.number.isRequired,
    active: PropTypes.bool.isRequired
  }).isRequired).isRequired,
  onChange: PropTypes.func.isRequired,
  onToggleClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  subjectsType: PropTypes.string.isRequired
}

export default SubjectList
