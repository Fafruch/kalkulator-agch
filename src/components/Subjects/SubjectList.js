import React from 'react'
import PropTypes from 'prop-types'
import { CSSTransitionGroup } from 'react-transition-group'

import Subject from './Subject'

const SubjectList = ({ subjects, onChange, onToggleClick, onDelete, subjectsType }) => {
  return (
    <ul className='list-unstyled'>
      <CSSTransitionGroup transitionName='subject' transitionEnterTimeout={300} transitionLeaveTimeout={300}>
        {subjects.map((subject, i) =>
          <li
            key={subject.id}
            className={subject.active ? 'subject-active' : 'subject-not-active'}
          >
            <Subject
              {...subject}
              onChange={onChange}
              onToggleClick={onToggleClick}
              onDelete={onDelete}
              subjectsType={subjectsType}
              iterator={i + 1}
            />
          </li>
        )}
      </CSSTransitionGroup>
    </ul>
  )
}

SubjectList.propTypes = {
  subjects: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    elementaryScore: PropTypes.number.isRequired,
    advancedScore: PropTypes.number.isRequired,
    active: PropTypes.bool.isRequired
  }).isRequired).isRequired,
  onChange: PropTypes.func.isRequired,
  onToggleClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  subjectsType: PropTypes.string.isRequired
}

export default SubjectList
