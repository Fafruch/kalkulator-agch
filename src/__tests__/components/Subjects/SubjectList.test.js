import React from 'react'
import { shallow } from 'enzyme'

import SubjectList from '../../../components/Subjects/SubjectList'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subjects: [
      {
        id: '0',
        name: 'Matematyka',
        elementaryScore: 100,
        advancedScore: 82,
        maxScore: {
          value: 182,
          formula: 'A/B'
        },
        active: true
      }
    ],
    subjectsType: 'primary'
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<SubjectList {...props} />)

  return {
    props,
    wrapper
  }
}

describe('SubjectList component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should render proper class when subject is not active', () => {
    const { wrapper } = setup({
      subjects: [
        {
          id: '0',
          name: 'Matematyka',
          elementaryScore: 100,
          advancedScore: 82,
          maxScore: {
            value: 182,
            formula: 'A/B'
          },
          active: false
        }
      ]
    })

    expect(wrapper.find('CSSTransitionGroup').shallow().find('li').is('.subject-not-active')).toEqual(true)
  })
})
