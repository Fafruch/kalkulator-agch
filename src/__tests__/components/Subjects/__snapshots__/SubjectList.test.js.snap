// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectList component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubjectList
    subjects={
        Array [
            Object {
              "active": true,
              "advancedScore": 82,
              "elementaryScore": 100,
              "id": "0",
              "maxScore": Object {
                "formula": "A/B",
                "value": 182,
              },
              "name": "Matematyka",
            },
          ]
    }
    subjectsType="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <CSSTransitionGroup
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={300}
        transitionLeave={true}
        transitionLeaveTimeout={300}
        transitionName="subject"
>
        <li
                className="subject-active"
        >
                <Connect(Subject)
                        subject={
                                Object {
                                        "active": true,
                                        "advancedScore": 82,
                                        "elementaryScore": 100,
                                        "id": "0",
                                        "maxScore": Object {
                                          "formula": "A/B",
                                          "value": 182,
                                        },
                                        "name": "Matematyka",
                                      }
                        }
                        subjectNumber={1}
                        subjectsType="primary"
                />
        </li>
</CSSTransitionGroup>,
      "className": "list-unstyled",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <li
            className="subject-active"
>
            <Connect(Subject)
                        subject={
                                    Object {
                                                "active": true,
                                                "advancedScore": 82,
                                                "elementaryScore": 100,
                                                "id": "0",
                                                "maxScore": Object {
                                                  "formula": "A/B",
                                                  "value": 182,
                                                },
                                                "name": "Matematyka",
                                              }
                        }
                        subjectNumber={1}
                        subjectsType="primary"
            />
</li>,
        ],
        "transitionAppear": false,
        "transitionEnter": true,
        "transitionEnterTimeout": 300,
        "transitionLeave": true,
        "transitionLeaveTimeout": 300,
        "transitionName": "subject",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Subject)
              subject={
                            Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
              }
              subjectNumber={1}
              subjectsType="primary"
/>,
            "className": "subject-active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "subject": Object {
                "active": true,
                "advancedScore": 82,
                "elementaryScore": 100,
                "id": "0",
                "maxScore": Object {
                  "formula": "A/B",
                  "value": 182,
                },
                "name": "Matematyka",
              },
              "subjectNumber": 1,
              "subjectsType": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": [Function],
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <CSSTransitionGroup
          transitionAppear={false}
          transitionEnter={true}
          transitionEnterTimeout={300}
          transitionLeave={true}
          transitionLeaveTimeout={300}
          transitionName="subject"
>
          <li
                    className="subject-active"
          >
                    <Connect(Subject)
                              subject={
                                        Object {
                                                  "active": true,
                                                  "advancedScore": 82,
                                                  "elementaryScore": 100,
                                                  "id": "0",
                                                  "maxScore": Object {
                                                    "formula": "A/B",
                                                    "value": 182,
                                                  },
                                                  "name": "Matematyka",
                                                }
                              }
                              subjectNumber={1}
                              subjectsType="primary"
                    />
          </li>
</CSSTransitionGroup>,
        "className": "list-unstyled",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <li
              className="subject-active"
>
              <Connect(Subject)
                            subject={
                                          Object {
                                                        "active": true,
                                                        "advancedScore": 82,
                                                        "elementaryScore": 100,
                                                        "id": "0",
                                                        "maxScore": Object {
                                                          "formula": "A/B",
                                                          "value": 182,
                                                        },
                                                        "name": "Matematyka",
                                                      }
                            }
                            subjectNumber={1}
                            subjectsType="primary"
              />
</li>,
          ],
          "transitionAppear": false,
          "transitionEnter": true,
          "transitionEnterTimeout": 300,
          "transitionLeave": true,
          "transitionLeaveTimeout": 300,
          "transitionName": "subject",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Subject)
                subject={
                                Object {
                                                "active": true,
                                                "advancedScore": 82,
                                                "elementaryScore": 100,
                                                "id": "0",
                                                "maxScore": Object {
                                                  "formula": "A/B",
                                                  "value": 182,
                                                },
                                                "name": "Matematyka",
                                              }
                }
                subjectNumber={1}
                subjectsType="primary"
/>,
              "className": "subject-active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "subject": Object {
                  "active": true,
                  "advancedScore": 82,
                  "elementaryScore": 100,
                  "id": "0",
                  "maxScore": Object {
                    "formula": "A/B",
                    "value": 182,
                  },
                  "name": "Matematyka",
                },
                "subjectNumber": 1,
                "subjectsType": "primary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": [Function],
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
